### PR TITLE
Build Infra Updates

### PR DESCRIPTION
**Pull request by kenotron**
In this change, we are re-organizing the project into a single package.

1. move everything back into src, the deprecating APIs will be available under satcheljs/lib/legacy/*
2. add tslint support
3. add jest support

### CODE REVIEW

**Comment by smikula** [comment]
Can we get rid of the /tools directory now?  (Or at least a bunch of it?)

File: package.json [hunk 1]
```diff
@@ -2,34 +2,48 @@
   "name": "satcheljs-suite",
```
Comment (smikula) [inline package.json]: >satcheljs-suite [](start = 11, length = 15)

This should just be `satcheljs` now.

File: package.json [hunk 2]
```diff
@@ -2,34 +2,48 @@
...
+    "test:start": "npm-run-all clean --parallel watch:*"
   },
   "devDependencies": {
     "@types/enzyme": "~2.8.1",
     "@types/jasmine": "~2.5.53",
-    "@types/jsdom": "~2.0.32",
+    "@types/jsdom": "^11.0.1",
     "@types/node": "~6.0.78",
-    "@types/react": "~15.0.27",
+    "@types/react": "^15.0.37",
```
Comment (smikula) [inline package.json]: >^ [](start = 21, length = 1)

I don't think we want a ^ dependency here.  This has bitten us in the past.

File: src/simpleAction.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    });
+
+    // Create a mutator
+    let simpleMutator = mutator(simpleActionCreator, function simpleActionMutator(actionMessage: any) {
+        target.apply(null, actionMessage.args);
+    });
+
+    // Return a function that dispatches that action
+    return (simpleActionCreator as any) as T;
+}
```
Comment (smikula) [inline src/simpleAction.ts]: This file should be deleted.  (It was replaced by simpleSubscribers.)

File: src/rootStore.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import { ObservableMap } from 'mobx';
 import { getGlobalContext } from './globalContext';
 
+export { ObservableMap };
```
Comment (smikula) [inline src/rootStore.ts]: Is this necessary?  Why didn't we need it before?

**Comment by kenotron** [comment]
YES!

---
In reply to: [314594927](https://github.com/Microsoft/satcheljs/pull/62#issuecomment-314594927) [](ancestors = 314594927)

File: src/simpleAction.ts (hunk 3 above)
Comment (kenotron) [inline src/simpleAction.ts]: :smile: 

---
In reply to: [126829826](https://github.com/Microsoft/satcheljs/pull/62#discussion_r126829826) [](ancestors = 126829826)

File: src/rootStore.ts (hunk 4 above)
Comment (kenotron) [inline src/rootStore.ts]: The no-unused-imports tslint rule was triggered on the ObservableMap, so I added an export.  I'm disabling that rule on that line instead.

---
In reply to: [126830331](https://github.com/Microsoft/satcheljs/pull/62#discussion_r126830331) [](ancestors = 126830331)

**Comment by smikula** [comment]
:shipit:

**Pull request merged**